middlewares/ip/ip_blacklist.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package ip
+
+import (
+	"net"
+
+	"github.com/chihaya/chihaya"
+	"github.com/chihaya/chihaya/config"
+	"github.com/chihaya/chihaya/errors"
+	"github.com/chihaya/chihaya/server/store"
+	"github.com/chihaya/chihaya/tracker"
+)
+
+func init() {
+	tracker.RegisterAnnounceMiddleware("ip_blacklist", blacklistAnnounceIP)
+}
+
+// ErrBlockedIP is returned by an announce middleware if any of the announcing IPs is disallowed
+var ErrBlockedIP = errors.NewMessage("disallowed IP address")
+
+// blacklistAnnounceIP provides a middleware that only allows IPs to announce that are not stored in an IPStore
```

Comment:
Go comments use 80 char line length.
Go code has no line length.
It's weird, but that's how the world works.